torch/csrc/jit/script/module.h

```diff
@@ -0,0 +1,256 @@
+    // if they were not already)
+    auto members_it = callee.member_inputs.begin();
+    for(size_t i = inputs.size(); i < fn->inputs().size(); i++) {
+      value_map[fn->inputs()[i]] = get_or_add_parameter(*members_it++);
+    }
+    for (auto* node : fn->nodes()) {
+      auto* new_node =
+          graph_->insertNode(graph_->createClone(node, value_map_func));
+      for (size_t i = 0; i < node->outputs().size(); ++i) {
+        value_map[node->outputs()[i]] = new_node->outputs()[i];
+      }
+    }
+
+    std::vector<Value*> outputs;
+    for (auto* output : fn->outputs()) {
+      outputs.push_back(value_map_func(output));
+    }
+    return outputs;
+  }
+  size_t num_inputs() const {
+    return graph_->inputs().size() - member_inputs.size();
+  }
+  Value * get_or_add_parameter(at::Tensor* slot) {
+    size_t first_parameter = graph()->inputs().size() - member_inputs.size();
+    for(size_t i = 0; i < member_inputs.size(); i++) {
+      // it is already a parameter
+      if(member_inputs[i] == slot) {
+        return graph()->inputs()[first_parameter + i];
+      }
+    }
+    // add it as a new parameter
+    member_inputs.push_back(slot);
+    return graph()->addInput();
+  }
+private:
+  std::string name_;
+  std::shared_ptr<Graph> graph_; // for debugging and for inlining
+  bool optimize;
+  GraphExecutor executor; // for execution
+  // member_inputs are a list of additional arguments appended to graph that are
+  // inputs that come from the members of the Module or its submodules.
+  // each is a pointer to a slot in the module that owns this Method or a submethod
+  // of the module.
+  // parameters and submodules can only be _added_ to script Modules to ensure
+  // these pointers always stay valid
+  std::vector<at::Tensor*> member_inputs;
+
+
+  // TODO: support that case where we allow _writes_ to parameters from
+  // compiled functions.
+  // This requires more sophisticated tracking of ssa values in Graphs so that
+  // stores to all modules can be lifted to the end of a graph execution.
+  // It also adds more complexity to adding actual module invocations
+  // to the executor, so currently it is not done.
+  // std::vector<at::Tensor*> member_outputs;
+
+  std::once_flag executor_init;
+};
+
+struct Module;
+
+struct NamedModule {
+  std::string name;
+  std::shared_ptr<Module> module;
+};
+
+struct NamedParameter {
+  NamedParameter(std::string name, at::Tensor tensor)
+  : name(std::move(name)), parameter(new at::Tensor(std::move(tensor))) {}
+
+  std::string name;
+  at::Tensor* slot() const {
+    return parameter.get();
+  }
+private:
+  // the extra level of indirection allows Methods to safely store pointers
+  // to the slots where parameters are kept while also allow parameters
+  // to be reassigned
+  std::unique_ptr<at::Tensor> parameter;
+};
+
+struct NamedMember {
+  enum Kind { Module, Parameter, Method, None };
+  // note: None is used to report undefined attributes;
+  Kind kind;
+  size_t offset;
```

Comment:
can you add some comment to this field? `offset` doesn't tell you anything about it